crates/relayer/src/chain/axon/msg.rs

```diff
@@ -0,0 +1,775 @@
+            source_channel: value.source_channel.as_str().parse().unwrap(),
+            destination_port: value.destination_port.as_str().parse().unwrap(),
+            destination_channel: value.destination_channel.as_str().parse().unwrap(),
+            data: value.data.as_ref().to_vec(),
+            timeout_height: TimeoutHeight::At(value.timeout_height.into()),
+            timeout_timestamp: Timestamp::from_nanoseconds(value.timeout_timestamp).unwrap(),
+        }
+    }
+}
+
+impl From<Packet> for contract::PacketData {
+    fn from(value: Packet) -> Self {
+        Self {
+            sequence: value.sequence.into(),
+            source_port: value.source_port.as_str().into(),
+            source_channel: value.source_channel.as_str().into(),
+            destination_port: value.destination_port.as_str().into(),
+            destination_channel: value.destination_channel.as_str().into(),
+            data: value.data.into(),
+            timeout_height: match value.timeout_height {
+                TimeoutHeight::At(h) => h.into(),
+                TimeoutHeight::Never => Default::default(),
+            },
+            timeout_timestamp: value.timeout_timestamp.nanoseconds(),
+        }
+    }
+}
+
+impl From<MsgConnectionOpenInit> for contract::MsgConnectionOpenInit {
+    fn from(value: MsgConnectionOpenInit) -> Self {
+        Self {
+            client_id: value.client_id.as_str().into(),
+            counterparty: value.counterparty.into(),
+            delay_period: value.delay_period.as_nanos() as u64,
+        }
+    }
+}
+
+impl TryFrom<Any> for contract::MsgConnectionOpenInit {
+    type Error = Error;
+
+    fn try_from(value: Any) -> Result<Self, Self::Error> {
+        Ok(MsgConnectionOpenInit::from_any(value)
+            .map_err(|e| Error::protobuf_decode(conn_open_init::TYPE_URL.into(), e))?
+            .into())
+    }
+}
+
+impl From<MsgConnectionOpenTry> for contract::MsgConnectionOpenTry {
+    fn from(value: MsgConnectionOpenTry) -> Self {
+        let (object_proof, client_proof, (consensus_proof, consensus_height), height) =
+            into_ethers_proofs(value.proofs);
+        Self {
+            client_id: into_ethers_client_id(Some(value.client_id)),
+            previous_connection_id: into_ethers_connection_id(value.previous_connection_id),
+            counterparty: value.counterparty.into(),
+            delay_period: value.delay_period.as_nanos() as u64,
+            counterparty_versions: value
+                .counterparty_versions
+                .into_iter()
+                .map(|v| v.into())
+                .collect(),
+            proof_height: height,
+            proof_init: object_proof,
+            proof_client: client_proof,
+            proof_consensus: consensus_proof,
+            consensus_height,
+            client_state: into_ethers_client_state(value.client_state),
+        }
+    }
+}
+
+impl TryFrom<Any> for contract::MsgConnectionOpenTry {
```

Comment:
It is an internal serialization. That is, worker serializes `MsgConnectionOpenTry` to `Any` and sends it by message sender, and then in chain endpoint it will be deserialized to concrete type and construct transaction. It has nothing to do with solidity definition of this type.

Reply:
I know there is nothing to do in solidity, but are `MsgConnectionOpenTry` or other familiar types implemented with protobuf format naturally, do we need to do something in Forcerelay?

Reply:
I just have added case to test `Any` conversion for `MsgConnectionOpenInit` and `MsgConnectionOpenTry`. Not add tests for all Msg types as the conversion are guaranteed by existing `From` implementation between `RawMsg` and `Msg`, and there have been already tests for those.